lib/PGcore.pm

```diff
@@ -549,6 +552,18 @@ sub store_persistent_data {    # will store strings only (so far)
 	$label;
 }
 
+sub update_persistent_data {    # will store strings only (so far)
+	my $self  = shift;
+	my $label = shift;
+	$self->{PERSISTENCE_HASH}->{$label} = join("", @_);
```

Comment:
The arguments should be unpacked on the first line (perlcritic says "Always unpack @_ first at line 555, column 1.  See page 178 of PBP.  (Severity: 4)").  The point is that this makes the expected arguments of the method much clearer.  Also, there is no need for the extra de-referencing arrow.  This should be
```suggestion
	my ($self, $label, @values) = @_;
	$self->{PERSISTENCE_HASH}{$label} = join('', @values);
```

Reply:
Thanks for all the comments and suggestions. All were adopted.
There is now also an additional hash `PERSISTENCE_HASH_UPDATED` used to signal to the front end which keys are new or have updated values. That will reduce unnecessary processing and database calls.